src/renderers/dom/shared/DOMProperty.js

```diff
@@ -111,30 +113,28 @@ var DOMPropertyInjection = {
         propName,
       );
 
-      if (__DEV__) {
-        DOMProperty.getPossibleStandardName[lowerCased] = propName;
-      }
-
       if (DOMAttributeNames.hasOwnProperty(propName)) {
         var attributeName = DOMAttributeNames[propName];
+
         propertyInfo.attributeName = attributeName;
-        if (__DEV__) {
-          DOMProperty.getPossibleStandardName[attributeName] = propName;
-        }
+
+        // Use the lowercase form of the attribute name to prevent
+        // badly cased React attribute alises from writing to the DOM.
+        DOMProperty.aliases[attributeName.toLowerCase()] = true;
```

Comment:
We need a good name for the "React way" attribute names.... We call them "special React properties", "javascript form", etc....